### PR TITLE
Use library for keccak, add subaddress class to namespace, delete src/sha3.php

### PR DESCRIPTION
**Pull request by recanman**

### CODE REVIEW

**Review by serhack — COMMENTED** [review]
TODO for us: it might be a good option to include at first some units tests about Keccak. I haven't had time to look for that library yet, but it seems more solid than the method `keccak_256` we had.

**Comment by recanman** [comment]
Yes, I'm planning on adding unit tests as part of my PHP8 port. I've tested this library manually, it works fine, and performance is around the same.

**Comment by BrianHenryIE** [comment]
A good thing about this, maybe unnoticed, is that the current [SHA3.php is GPL licenced,](https://github.com/monero-integrations/monerophp/blob/099c2fef2ecce4142c56725f302cc36e01d362c2/src/SHA3.php#L22) but [kornrunner/php-keccak is MIT](https://github.com/kornrunner/php-keccak/blob/main/LICENSE). I suspect (IANAL) that currently, monerophp should not be labelled as MIT, but this will solve that.

**Comment by recanman** [comment]
May I get a review on this PR? Thanks.

File: src/subaddress.php [hunk 1]
```diff
@@ -70,9 +76,9 @@ private function ge_scalarmult_base($scalar)
 	 */
 	private function keccak_256($message)
 	{
-		$keccak256 = SHA3::init(SHA3::KECCAK_256);
-		$keccak256->absorb(hex2bin($message));
-		return bin2hex($keccak256->squeeze(32)) ;
+		$message_bin = hex2bin($message);
+        $hash = keccak::hash($message_bin, 256);
```
Comment (serhack) [inline src/subaddress.php:80]: I'm seeing something weird with indentation, is it correct? It does not seem so.

Comment (recanman) [reply src/subaddress.php:80]: It confused me as well. It was a mix of tabs and spaces.

Comment (serhack) [reply src/subaddress.php]: Thanks for solving it!

**Comment by recanman** [comment]
I've fixed the issue. There was also a merge conflict, and I fixed that as well.
Anything else @serhack?

**Comment by serhack** [comment]
Looks good, thanks for the force push!

**Pull request merged**